src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurRequestExtractFileServiceImpl.java

```diff
@@ -0,0 +1,172 @@
+
+public class ConcurRequestExtractFileServiceImpl implements ConcurRequestExtractFileService {
+    private static org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(ConcurRequestExtractFileServiceImpl.class);
+    protected String acceptedDirectoryName;
+    protected String rejectedDirectoryName;
+    protected BatchInputFileType batchInputFileType;
+    protected BatchInputFileService batchInputFileService;
+    protected ConcurRequestExtractFileValidationService concurRequestExtractFileValidationService;
+
+    public List<String> getUnprocessedRequestExtractFiles() {
+        List<String> fileNamesToLoad = getBatchInputFileService().listInputFileNamesWithDoneFile(getBatchInputFileType());
+        return fileNamesToLoad;
+    }
+
+    public boolean processFile(String requestExtractFileName) {
+        boolean processingSuccess = true;
+        Object parsedFile;
+        try {
+            parsedFile = loadFile(requestExtractFileName);
+            processingSuccess = requestExtractHeaderRowValidatesToFileContents(parsedFile);
+        } catch (RuntimeException re) {
+            LOG.error("Caught exception trying to load: " + requestExtractFileName, re);
+            processingSuccess = false;
+        }
+        return processingSuccess;
+    }
+
+    public boolean requestExtractHeaderRowValidatesToFileContents(Object parsedFile) {
+        boolean headerValidationPassed = false;
+        List<ConcurRequestExtractFile> requestExtractFiles = (ArrayList<ConcurRequestExtractFile>) parsedFile;
+        if (requestExtractFiles.isEmpty() || requestExtractFiles.size() != 1) {
+            LOG.error("Header validation should have received single parsed file. More or less than one parsed file was detected.");
+        }
+        else {
+            ConcurRequestExtractFile requestExtractFile = requestExtractFiles.get(0);
+            LOG.info("batchDate=" + requestExtractFile.getBatchDate() + "=   recordCount=" + requestExtractFile.getRecordCount() + "=     totalApprovedAmount=" + requestExtractFile.getTotalApprovedAmount() + "=");
+            headerValidationPassed = (getConcurRequestExtractFileValidationService().fileRowCountMatchesHeaderRowCount(requestExtractFile) &&
+                                      getConcurRequestExtractFileValidationService().fileApprovedAmountsMatchHeaderApprovedAmount(requestExtractFile));
+        }
+        return headerValidationPassed;
+    }
+
+    public void performRejectedRequestExtractFileTasks(String pathAndFileName) {
+        moveFile(pathAndFileName, getRejectedDirectoryName());
+    }
+
+    public void performAcceptedRequestExtractFileTasks(String pathAndFileName) {
+        moveFile(pathAndFileName, getAcceptedDirectoryName());
+    }
+
+    public void performDoneFileTasks(String requestExtractFileName) {
+        File doneFile = new File(StringUtils.substringBeforeLast(requestExtractFileName, ".") + ".done");
```

Comment:
I know each batch service we have defines their own method to remove these .done files but I am wondering if you could use some generic service to do this. I see a new FileStorageService was created in 7.x and I wonder if it would make more sense to use that?